test/tools/router_check/router.cc

```diff
@@ -100,19 +100,14 @@ RouterCheckTool::RouterCheckTool(
 bool RouterCheckTool::compareEntriesInJson(const std::string& expected_route_json) {
   Json::ObjectSharedPtr loader = Json::Factory::loadFromFile(expected_route_json, *api_);
   loader->validateSchema(Json::ToolSchema::routerCheckSchema());
-
   bool no_failures = true;
   for (const Json::ObjectSharedPtr& check_config : loader->asObjectArray()) {
     headers_finalized_ = false;
     ToolConfig tool_config = ToolConfig::create(check_config);
     tool_config.route_ = config_->route(*tool_config.headers_, tool_config.random_value_);
-
     std::string test_name = check_config->getString("test_name", "");
-    if (details_) {
-      std::cout << test_name << std::endl;
-    }
+    tests_.push_back(std::pair<std::string, std::vector<std::string>>(test_name, {}));
```

Comment:
nit: I think you can do `tests_.emplace_back(test_name, {});` (If that doesn't work don't spend time on it). Same below.